File: financial-management-platform/financial-api/Models/Transaction.cs
namespace FinancialApi.Models;

public class Transaction
{
    public int Id { get; set; }
    public string Type { get; set; } = "expense"; // income | expense
    public int? CategoryId { get; set; }
    public Category? Category { get; set; }
    public string PaymentMethod { get; set; } = string.Empty; // bank | cash
    public decimal Amount { get; set; }
    public string Description { get; set; } = string.Empty;
    public int UserId { get; set; }
    public User User { get; set; } = null!;
    public DateTime CreatedAt { get; set; } = DateTime.UtcNow;
    public DateTime? ModifiedAt { get; set; }
}

File: financial-management-platform/financial-dashboard-ios/FinancialDashboard/Models/Transaction.swift
import Foundation

// MARK: - Transaction (expense)

struct Transaction: Codable, Identifiable {
    var id: Int
    var date: String        // yyyy-MM-dd (user-selected date)
    var createdAt: String   // ISO 8601 datetime (server timestamp)
    var description: String
    var categoryId: Int?
    var amount: Double
    var paymentMethod: String  // "bank" | "cash"
    var type: String           // "expense"
}

// MARK: - Income record

struct IncomeRecord: Codable, Identifiable {
    var id: Int
    var date: String
    var createdAt: String
    var description: String
    var amount: Double
    var paymentMethod: String  // "bank" | "cash" | "withdrawal"
}

// MARK: - Request models

struct CreateTransactionRequest: Encodable {
    var date: String
    var description: String
    var categoryId: Int?
    var amount: Double
    var paymentMethod: String
    var type: String = "expense"
}

struct UpdateTransactionRequest: Encodable {
    var date: String
    var description: String
    var categoryId: Int?
    var amount: Double
    var paymentMethod: String
    var type: String = "expense"
}

struct CreateIncomeRequest: Encodable {
    var date: String
    var description: String
    var amount: Double
    var paymentMethod: String
}

struct UpdateIncomeRequest: Encodable {
    var date: String
    var description: String
    var amount: Double
    var paymentMethod: String
}
